src/main/java/org/wso2/carbon/connector/amazonsqs/connection/SqsConnection.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.carbon.connector.amazonsqs.connection;
+
+import org.wso2.carbon.connector.amazonsqs.object.SqsClientManager;
+import org.wso2.carbon.connector.core.connection.Connection;
+import software.amazon.awssdk.services.sqs.SqsClient;
+
+/**
+ * This class manages connections to Amazon SQS using the AWS SDK.
+ */
+public class SqsConnection implements Connection {
```

Comment:
We should bind a SqsClient per each connection. Therefore we may need to have singleton instance of SqsClient here.

Reply:
Already added the client using the singleton instance here. https://github.com/wso2-extensions/esb-connector-amazonsqs/pull/40/files#diff-336c4599b0ac3b1c397ec601c19ef576703af1a7050863d652f36e741cece7ecR34

Reply:
Since we have used a singleton `SqsClientManager` it will return the same SQSClient for two connections.

Reply:
Updated as we discussed offline.